Services/Storage/StorageRecords.cs

```diff
@@ -0,0 +1,371 @@
+            IInstance instance,
+            IConcurrencyConfig concurrencyConfig)
+        {
+            this.log = logger;
+            this.instance = instance;
+            this.docDb = docDb;
+            this.disposedValue = false;
+            this.storageConfig = null;
+            this.client = null;
+            this.concurrencyConfig = concurrencyConfig;
+        }
+
+        public IStorageRecords Init(StorageConfig cfg)
+        {
+            this.instance.InitOnce();
+
+            this.storageConfig = cfg;
+
+            // Used only for logging
+            this.storageName = cfg.DocumentDbDatabase + "/" + cfg.DocumentDbCollection;
+
+            this.instance.InitComplete();
+
+            return this;
+        }
+
+        public async Task<StorageRecord> GetAsync(string id)
+        {
+            await this.SetupStorageAsync();
+
+            try
+            {
+                this.log.Debug("Fetching record...", () => new {this.storageName, id});
+                var response = await this.docDb.ReadAsync(this.client, this.storageConfig, id);
+                this.log.Debug("Record fetched", () => new { this.storageName, id });
+
+                var record = StorageRecord.FromDocumentDb(response?.Resource);
+
+                if (record.IsExpired())
+                {
+                    this.log.Debug("The resource requested has expired, deleting...", () => new { this.storageName, id });
+                    await this.TryToDeleteExpiredRecord(id);
+                    this.log.Debug("Expired resource deleted", () => new { this.storageName, id });
+
+                    throw new ResourceNotFoundException($"The resource {id} doesn't exist.");
+                }
+
+                return record;
+            }
+            catch (DocumentClientException e) when (e.StatusCode == HttpStatusCode.NotFound)
+            {
+                this.log.Debug("The resource requested doesn't exist.", () => new{this.storageName, id});
+                throw new ResourceNotFoundException($"The resource {id} doesn't exist.");
+            }
+        }
+        public async Task<bool> ExistsAsync(string id)
+        {
+            await this.SetupStorageAsync();
+
+            try
+            {
+                var response = await this.docDb.ReadAsync(this.client, this.storageConfig, id);
+                var record = StorageRecord.FromDocumentDb(response?.Resource);
+
+                if (record.IsExpired())
+                {
+                    this.log.Info("The resource requested has expired.", () => new { this.storageName, id });
+                    await this.TryToDeleteExpiredRecord(id);
+                    return false;
+                }
+
+                return true;
+            }
+            catch (DocumentClientException e) when (e.StatusCode == HttpStatusCode.NotFound)
+            {
+                return false;
+            }
+        }
+        public async Task<IEnumerable<StorageRecord>> GetAllAsync()
```

Comment:
missing empty line, check all doc (e.g. in the IDE run "format document)

Reply:
Fixed.